### ISSUE TITLE
Wrong assumption about require.cache

### ISSUE THREAD

**Issue by tbassetto** [ivogabe/gulp-typescript#315]
I am working on a big project where suddenly it started failing after adding `eslint-config-angular` to the build step. But I dug and I think the issue could have appeared with other modules for other reasons.

This is the error when compiling TypeScript files:
```
            if (require.cache[i].exports === typescript) {
                                ^

TypeError: Cannot read property 'exports' of undefined
    at Function.Host.getLibDefault (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-typescript/release/host.js:57:33)
    at Host.getSourceFile (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-typescript/release/host.js:18:29)
    at findSourceFile (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-typescript/node_modules/typescript/lib/typescript.js:40198:29)
    at processSourceFile (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-typescript/node_modules/typescript/lib/typescript.js:40146:27)
    at processRootFile (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-typescript/node_modules/typescript/lib/typescript.js:40056:13)
    at Object.createProgram (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-typescript/node_modules/typescript/lib/typescript.js:39608:17)
    at ProjectCompiler.inputDone (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-typescript/release/compiler.js:66:48)
    at CompileStream.end (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-typescript/release/main.js:53:31)
    at DestroyableTransform.onend (/Users/thobasse/Code/wx2-admin-web-client/node_modules/gulp-sourcemaps/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:495:10)
    at DestroyableTransform.g (events.js:260:16)
```

The line crashing is this one:
https://github.com/ivogabe/gulp-typescript/blob/22c8be234895c0d94ba9757b6fc65817e37275f8/release/host.js#L49

I have added more logs before the line failing and this the result:
```
Object.prototype.hasOwnProperty.call(require.cache, i) => true
require.cache.hasOwnProperty(i) => true
i in require.cache => true
require.cache[i] => undefined
i => […]/node_modules/eslint-config-angular/index.js
```

So it looks like we can have an entry in `require.cache` that is there but is also `undefined`… I am running node 4.4.1 and I don't know node enough to know how we can end up like this. But maybe the check on line 47 (https://github.com/ivogabe/gulp-typescript/blob/22c8be234895c0d94ba9757b6fc65817e37275f8/release/host.js#L47) should also looks for `.exports` and not just the presence of `i`.

**Comment by ivogabe** [comment]
Thanks for reporting & debugging! It should be fixed in commit ebed5bf59fc07bb5e4f1c329331cc7a4aad32ae6, I will update a new version soon but you can already test it by downloading the repository from GitHub.

**Comment by tbassetto** [comment]
I haven't tested the code but it won't fix it :) It's `require.cache[i]` that can be `undefined`, not `require.cache[i].exports`. So `require.cache[i].exports === undefined` will still trigger a `Uncaught TypeError: Cannot read property 'exports' of undefined`. The check should be `require.cache[i] === undefined`.

How `require.cache['some file']` can be `undefined`, I have no idea though. I don't know node's internals good enough.

**Comment by ivogabe** [comment]
Oops, you're right. Should be now fixed in 9eeff838a97f196897afa23587044b71e7ee230b
